CMakeLists.txt

```diff
@@ -42,56 +45,91 @@ add_definitions(-DGOOGLE_STRIP_LOG=1)
 # add_definitions(-DEIGEN_DEFAULT_TO_ROW_MAJOR)
 add_definitions(-DEIGEN_INITIALIZE_MATRICES_BY_ZERO)
 
-find_package(OpenCV REQUIRED)
-find_package(Python3 REQUIRED Interpreter Development)
+set(THIRD_PARTY_DIR "${CMAKE_SOURCE_DIR}/thirdparty")
+list(APPEND CMAKE_PREFIX_PATH ${THIRD_PARTY_DIR})
 
-link_directories(
-        ${PROJECT_SOURCE_DIR}/lib
-        ${PROJECT_SOURCE_DIR}/thirdparty/gflags/lib
-        ${PROJECT_SOURCE_DIR}/thirdparty/glog/lib
-        ${PROJECT_SOURCE_DIR}/thirdparty/Pangolin/lib
-        ${PROJECT_SOURCE_DIR}/thirdparty/jsoncpp/lib
-        ${PROJECT_SOURCE_DIR}/thirdparty/gperftools/lib
-        ${PROJECT_SOURCE_DIR}/thirdparty/DBoW2/lib
-        ${PROJECT_SOURCE_DIR}/thirdparty/pnp/build
-        ${PROJECT_SOURCE_DIR}/thirdparty/ceres-solver/lib
-        /usr/local/lib
+find_package(OpenCV REQUIRED PATHS ${THIRD_PARTY_DIR}/opencv/install)
```

Comment:
We should add the following above this block so that CMake doesn't accidentally link to a version that was installed onto the OS or exported from another project:
```cmake
+set(Eigen3_DIR ${THIRD_PARTY_DIR}/eigen/share/eigen3/cmake)
+set(Ceres_DIR ${THIRD_PARTY_DIR}/ceres-solver/lib/cmake/Ceres)
+set(gflags_DIR ${THIRD_PARTY_DIR}/gflags/lib/cmake/gflags)
+set(glog_DIR ${THIRD_PARTY_DIR}/glog/lib/cmake/glog)
+set(gtest_DIR ${THIRD_PARTY_DIR}/googletest/lib/cmake/GTest)
+set(jsoncpp_DIR ${THIRD_PARTY_DIR}/jsoncpp/lib/cmake/jsoncpp)
+set(Pangolin_DIR ${THIRD_PARTY_DIR}/Pangolin/lib/cmake/Pangolin)
+set(DBoW2_DIR ${THIRD_PARTY_DIR}/DBoW2/lib/cmake/DBoW2)
```